javaslang/src/main/java/javaslang/collection/Seq.java

```diff
@@ -1153,4 +1149,38 @@ default boolean startsWith(Iterable<? extends T> that, int offset) {
     default Function1<Integer, T> withDefault(Function<? super Integer, ? extends T> defaultFunction) {
         return i -> (i >= 0 && i < length()) ? apply(i) : defaultFunction.apply(i);
     }
+
+    default java.util.List<T> asUnmodifiableList() { return new SeqList<>(Seq.this); }
```

Comment:
I know there's a danger about arguing about names forever, but perhaps "Java" should be in the name here, for consistency with `toJavaList()`.

What about `toImmutableJavaList()` or `asImmutableJavaList()`? 

A name starting with "to" should be a little easier to discover, especially when looking for `toJavaList()`. Although perhaps "as" helps to suggest a cheap "view" of the underlying list rather than an expensive "copy", which might be why I suggested `asJavaList()` in https://github.com/javaslang/javaslang/issues/1728. Meh. Either way.

Personally I think the word "immutable" sounds better than "unmodifiable". Plus Guava's `ImmutableList`s have some nice properties which these lists will share, whereas Java's `unmodifiableList`s... don't (for one thing, they aren't really immutable).

https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#unmodifiableList(java.util.List)